lib/broker_discovery.dart

```diff
@@ -38,10 +38,11 @@ class BrokerDiscoveryClient {
     try {
       for (var interface in interfaces) {
         try {
-          _socket.joinMulticast(new InternetAddress("239.255.255.230"), interface: interface);
-        } catch (e) {
-          _socket.joinMulticast(new InternetAddress("239.255.255.230"), interface: interface);
-        }
+          _socket.joinMulticast(
+            new InternetAddress("239.255.255.230"),
+            interface
```

Comment:
Dart used to have `interface` as a named parameter. This catches the error generated when using those versions of Dart. joinMulticast() also can fail because the interface is a stubbed network interface, that case is now handled.